### PR TITLE
BDD tests for maintenance

### PR DESCRIPTION
**Pull request by dofinn**
```
Running Suite: Maintenance Suite
================================
Random Seed: 1594099769
Will run 7 of 7 specs

•••••••
Ran 7 of 7 Specs in 0.000 seconds
SUCCESS! -- 7 Passed | 0 Failed | 0 Pending | 0 Skipped
PASS
coverage: 70.4% of statements

Ginkgo ran 1 suite in 2.453542127s
Test Suite Passed
```

### CODE REVIEW

File: pkg/maintenance/maintenance_test.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+		testComment           = "test comment"
+		testOperatorName      = "managed-upgrade-operator"
+		testCreatedByOperator = testOperatorName
+		testCreatedByTest     = "Tester the Creator"
+		testNow               = strfmt.DateTime(time.Now().UTC())
+		testEnd               = strfmt.DateTime(time.Now().UTC().Add(90 * time.Minute))
+		testVersion           = "V-1.million.25"
+		testGettableSilences  amv2Models.GettableSilences
+
+		// Create test silence NOT created by the operator
```
Comment (mrbarge) [inline pkg/maintenance/maintenance_test.go:35]: Minor nit, this comment seems to indicate the opposite of what's being created.

**Comment by mrbarge** [comment]
Other than one trivial noted thing, this looks great to me. I'll let more :eyes: have an opportunity to see it before lgtm'ing it.

File: pkg/maintenance/alertManagerSilenceClient.go [hunk 2]
```diff
@@ -9,12 +9,19 @@ import (
 	"net/http"
 )
 
+//go:generate mockgen -destination=alertManagerSilenceMock.go -package=maintenance github.com/openshift/managed-upgrade-operator/pkg/maintenance AlertManagerSilencer
```
Comment (mrbarge) [inline pkg/maintenance/alertManagerSilenceClient.go:12]: If I recall, was the reason for not putting it the 'mocks' package dir due to the cyclic import that results in?

Comment (dofinn) [reply pkg/maintenance/alertManagerSilenceClient.go:12]: correct

File: pkg/maintenance/maintenance_test.go (hunk 1 above)
Comment (dofinn) [reply pkg/maintenance/maintenance_test.go:35]: So my comment is incorrect. This was a validation that the operator will only retrieve maintenance's created by the operator. The actual test got moved to another spot in the file. Ill ammend

**Pull request merged**